test/collection_serializer_test.rb

```diff
@@ -23,6 +33,11 @@ def build_named_collection(*resource)
         resource
       end
 
+      def build_named_collection_with_matching_serializer(*resource)
+        resource.define_singleton_method(:name) { 'SerializedResource' }
```

Comment:
do you have a real-world example of this?  Previously there was a test for a monkey-patched array instance and it made me unsure if this was a real feature